server/api/notificationSettings.js

```diff
@@ -62,16 +62,23 @@ router.post("notificationSettings.unsubscribe", async (ctx) => {
   ctx.assertUuid(id, "id is required");
   ctx.assertPresent(token, "token is required");
 
-  const setting = await NotificationSetting.findByPk(id);
-  if (setting) {
-    if (token !== setting.unsubscribeToken) {
-      ctx.redirect(`${process.env.URL}?notice=invalid-auth`);
-    }
+  const setting = await NotificationSetting.findByPk(id, {
+    include: [
+      {
+        model: Team,
+        required: true,
+        as: "team",
+      },
+    ],
+  });
 
+  if (setting && setting.unsubscribeToken === token) {
     await setting.destroy();
+    ctx.redirect(`${setting.team.url}/settings/notifications?success`);
```

Comment:
fix: Only tangentially related fix – I noticed we were redirecting to the main domain instead of the correct team subdomain here.